### PR TITLE
Special handling of package:test on install tab

### PR DESCRIPTION
**Pull request by isoos**
#2125

### CODE REVIEW

**Comment by kevmoo** [comment]
This seems like something that should be generalized.

pkg: build_runner, build_web_compilers, build_test, build_verify, test_descriptor, test_process

In general, all of these should be shown as dev dependencies.

We could even use our own stats to drive this, right? Either from pubspecs of published packages or telemetry on the pub site

**Comment by isoos** [comment]
> pkg: build_runner, build_web_compilers, build_test, build_verify, test_descriptor, test_process

Added these. Anything else to immediately add?

> This seems like something that should be generalized. We could even use our own stats to drive this, right?

Agreed. However, we have higher priority items, so let's start with the custom whitelist.

**Comment by jonasfj** [comment]
> This seems like something that should be generalized.

I agree, as I suggested in #2125 I would much prefer if we just rendered `INSTALL.md` on the install tab, if the package contains an `INSTALL.md` file (exact file name match).

I doubt this would be hard to do, we just need to patch pana, update the model, and fork `backfill_example.dart` to backfull `INSTALL.md`. This way it's also useful for packages that are globally activated, like `mono_repo` or packages with external dependencies like things using `dart:ffi`.

File: app/lib/package/overrides.dart [hunk 1]
```diff
@@ -19,6 +19,17 @@ const redirectDartdocPages = <String, String>{
   'flutter': 'https://docs.flutter.io/',
 };
 
+/// Known packages that should be put in `dev_dependencies`
+const devDependencyPackages = <String>{
```
Comment (jonasfj) [inline app/lib/package/overrides.dart]: Can we leave a comment here stating that this is a temporary hack that will be removed when:
https://github.com/dart-lang/pub-dev/issues/3403 is probably implemented?

**Pull request merged**